### ISSUE TITLE
Payment broken in Magento 2.0.4

### ISSUE THREAD

**Issue by thomvanderboon** [magento/magento2#4076]
## Steps to reproduce
1. Install Magento 2.0.4 branch.
2. Configure Paypal or External payent plugin
3. buy a product, select a payment method
4. Circling wheel

Paypal is broken (See https://github.com/magento/magento2/issues/4064 ), Pay.nl plugin is broken, Multisafepay.nl plugin is broken
## Expected result
1. Go to payment on external site
2. ...
## Actual result
1. Wheel keeps circling for ever (I aborted it after 3 hours)
2. Apache has to start more workers (for some reason the I think the sessions keep hanging)
3. Server becomes sluggish (reboot needed)

**Comment by bgabrovsek** [comment]
Somebody should really provide a fix for this.
I have an online store and nobody can pay through it.

**Comment by thomvanderboon** [comment]
@bgabrovsek Did you upgrade Magento 2.0.2 -> 2.0.4 or 2.0.2 -> 2.0.3 -> 2.0.4 ?

**Comment by bgabrovsek** [comment]
I am still on 2.0.2.

In the debug console:

**IE returns:** 

```
XMLHttpRequest: Network Error 0x2f78, Could not complete the operation due to error 00002f78.`
```

**Chrome returns**: 

```
Failed to load resource: net::ERR_CONNECTION_RESET
/checkout/#payment:1 Uncaught SyntaxError: Unexpected end of input
```

**Mozilla returns:**

```
SyntaxError: JSON.parse: unexpected end of data at line 1 column 1 of the JSON data
.process()  error-processor.js:19
<anonimno> set-payment-method.js:51
jQuery.Callbacks/fire() jquery.js:3099
jQuery.Callbacks/self.fireWith() jquery.js:3211
done() jquery.js:9312
.send/callback() jquery.js:9720
```

.. and all of them return a warning "jQuery.browser is deprecated".

**Comment by thomvanderboon** [comment]
Just tested on development branch. Same problem. :(

**Comment by YevSent** [comment]
Hi @thomvanderboon could you provide more details? Something like browser error log, exception log,  debug log, etc, because we can't reproduce your issue.

**Comment by thomvanderboon** [comment]
What I also noted is that the moment this happens:

In my processes I see the following entries (ps aux):

Before:

root     12657  0.0  0.1 378272 15256 ?        Ss   10:19   0:00 /usr/sbin/httpd
apache   12659  0.0  0.0 378272  7772 ?        S    10:19   0:00 /usr/sbin/httpd
apache   12660  0.0  0.0 378272  7772 ?        S    10:19   0:00 /usr/sbin/httpd
apache   12661  0.0  0.0 378272  7792 ?        S    10:19   0:00 /usr/sbin/httpd
apache   12662  0.0  0.0 378272  7772 ?        S    10:19   0:00 /usr/sbin/httpd
apache   12663  0.0  0.0 378272  7772 ?        S    10:19   0:00 /usr/sbin/httpd
apache   12664  0.0  0.0 378272  7772 ?        S    10:19   0:00 /usr/sbin/httpd
apache   12665  0.0  0.0 378272  7772 ?        S    10:19   0:00 /usr/sbin/httpd
apache   12666  0.0  0.0 378272  7772 ?        S    10:19   0:00 /usr/sbin/httpd

after

root     12657  0.0  0.1 378272 15256 ?        Ss   10:19   0:00 /usr/sbin/httpd
apache   12659  1.0  1.1 553104 92248 ?        S    10:19   0:01 /usr/sbin/httpd
apache   12660  0.5  0.5 412912 48320 ?        S    10:19   0:00 /usr/sbin/httpd
apache   12661  1.5  1.0 444652 80700 ?        S    10:19   0:02 /usr/sbin/httpd
apache   12662  0.4  0.7 423260 58356 ?        S    10:19   0:00 /usr/sbin/httpd
apache   12663  1.2  0.9 441244 77188 ?        S    10:19   0:02 /usr/sbin/httpd
apache   12664  3.1  1.5 583736 122516 ?       S    10:19   0:05 /usr/sbin/httpd
apache   12665  0.9  0.9 443204 79084 ?        S    10:19   0:01 /usr/sbin/httpd
apache   12666  0.2  0.4 401048 34852 ?        S    10:19   0:00 /usr/sbin/httpd
apache   12726  1.1  0.4 397432 32872 ?        S    10:21   0:00 /usr/sbin/httpd
apache   12767  0.0  0.0  80932  3380 ?        S    10:22   0:00 /usr/sbin/sendmail -t -i
apache   12768  0.0  0.0  80900  3376 ?        S    10:22   0:00 /usr/sbin/postdrop -r

The sendmail and postdrop messages stay active for over one day

No entries in debug.log, system.log or exception.log

**Comment by thomvanderboon** [comment]
A little more digging in the maillog 

pr 11 10:52:08 shop postfix/postdrop[13312]: warning: mail_queue_enter: create file maildrop/800407.13312: Permission denied
Apr 11 10:52:18 shop postfix/postdrop[13312]: warning: mail_queue_enter: create file maildrop/800589.13312: Permission denied
Apr 11 10:52:28 shop postfix/postdrop[13312]: warning: mail_queue_enter: create file maildrop/800778.13312: Permission denied
Apr 11 10:52:38 shop postfix/postdrop[13312]: warning: mail_queue_enter: create file maildrop/800961.13312: Permission denied
Apr 11 10:52:48 shop postfix/postdrop[13312]: warning: mail_queue_enter: create file maildrop/801199.13312: Permission denied

A little more testing and found the permissions on both /usr/sbin/postqueue and /usr/sbin/postdrop to be incorrect.

So I now know how to reproduce (on CentOS 6.7):

service postfix stop
chmod 777 /usr/sbin/postqueue
chmod 777 /usr/sbin/postdrop
service postfix start

How to solve the problem (on CentOS 6.7):

service postfix stop
chgrp postdrop /usr/sbin/postqueue
chgrp postdrop /usr/sbin/postdrop
chmod g+s /usr/sbin/postqueue
chmod g+s /usr/sbin/postdrop
service postfix stop

I don't know how these files got the wrong permissions.

**Comment by YevSent** [comment]
Any errors in browser console when you click on `Continue to PayPal` button?

**Comment by thomvanderboon** [comment]
Please see above. 2 days ago post by @bgabrovsek

**Comment by YevSent** [comment]
@thomvanderboon could you specify additional details, like browser call for this error or some additional configuration, we tried different test cases and configurations and issue is not reproducible.

**Comment by asemenenko** [comment]
@joni-jones, @thomvanderboon @bgabrovsek 
issue is already resolved in MAGETWO-48387
It will be available with the next version update. Commit https://github.com/magento/magento2/commit/32ca5c97304a1bd84cfbee7cec3d57c9307da9a6

**Comment by thomvanderboon** [comment]
Shall I close this issue?

**Comment by asemenenko** [comment]
@thomvanderboon you mentioned that it is reproduced on the dev branch, can you re-check?

**Comment by maksek** [comment]
Closing issue due no activity for long time. Issue was fixed in develop Commit 32ca5c9.

**Comment by KrystynaKabannyk** [comment]
Hello @thomvanderboon, this issue has been fixed in the 2.0.7 patch release. Thanks for reporting the issue! If you have additional questions or proposals you are welcome to create another one.
